staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -276,6 +240,20 @@ func (c *AvailableConditionController) sync(key string) error {
 	}
 	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
 	if apiService.Spec.Service != nil && c.serviceResolver != nil {
+		restConfig, err := c.dynamicRestConfigProvider.GetRestConfig()
+		if err != nil {
+			return err
+		}
+		proxyTransport, err := rest.TransportFor(restConfig)
+		if err != nil {
+			return err
+		}
+		discoveryClient := &http.Client{
```

Comment:
creating an http client on every `sync` doesn't seem right, we should reuse it if `restConfig` hasn't changed.